FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -138,55 +118,68 @@ extension ComposeViewController {
     @objc private func handleSendTap() {
         sendMsgTapHandler()
     }
-
-    private func dismissKeyboard() {
-        view.endEditing(true)
-    }
-
-    @objc private func convertStringToLowercase(textField _: UITextField) {
-        txtRecipient.text = txtRecipient.text?.lowercased()
-    }
 }
 
+// MARK: - Message Sending
+
 extension ComposeViewController {
     private func sendMsgTapHandler() {
-        dismissKeyboard()
-
-        guard isInputValid(),
-            let email = txtRecipient.text,
-            let text = txtMessage.text
-        else { return }
-
-        let pgpText = "compose_missed_public".localized
-        let subject = viewModel.isReply
-            ? "Re: \(viewModel.replyToSubject ?? "(no subject)")"
-            : txtSubject.text ?? "(no subject)"
-        let message = viewModel.isReply
-            ? "compose_reply_successfull".localized
-            : "compose_sent".localized
+        view.endEditing(true)
+        guard isInputValid() else { return }
 
         showSpinner("sending_title".localized)
 
         Promise<Void> { [weak self] in
+            try await(self!.sendMessage())
+        }.then(on: .main) { [weak self] in
+            self?.handleSuccessfullySentMessage()
+        }.catch(on: .main) { [weak self] error in
+            self?.showAlert(error: error, message: "compose_error".localized)
+        }
+    }
+
+    private func sendMessage() -> Promise<Void> {
+        Promise { [weak self] in
             guard let self = self else { return }
+            guard let email = self.contextToSend.resipient, let text = self.contextToSend.message else {
+                assertionFailure("Text and Email should not be nil at this point. Fail in checking");
+                return
+            }
+
+            let subject = self.input.subjectReplyTitle
+                ?? self.contextToSend.subject
+                ?? "(no subject)"
+
+            let pgpText = "compose_missed_public".localized
+
             let lookupRes = try await(self.attesterApi.lookupEmail(email: email))
-            guard let recipientPubkey = lookupRes.armored else { return self.showAlert(message: pgpText) }
-            let realm = try Realm() // TODO: Anton - Refactor to use db service
 
-            guard let myPubkey = realm.objects(KeyInfo.self).map({ $0.public }).first else { return self.showAlert(message: pgpText) }
-            let encrypted = self.encryptMsg(pubkeys: [myPubkey, recipientPubkey], subject: subject, message: text, email: email)
+            guard let recipientPubkey = lookupRes.armored else {
+                return self.showAlert(message: pgpText)
+            }
+
+            guard let myPubkey = self.storageService.publicKey() else {
+                return self.showAlert(message: pgpText)
```

Comment:
alert: `Could not find any suitable Public Key on your account. If FlowCrypt app well set up?`